apps/prairielearn/assets/scripts/instructorAssessmentManualGradingInstanceQuestion.js

```diff
@@ -89,10 +89,13 @@ function resetInstructorGradingPanel() {
       form.querySelectorAll('.js-auto-score-value-info').forEach((element) => {
         element.style.display = 'none';
       });
-      form.querySelectorAll('.js-auto-score-value-input').forEach((input) => {
-        input.classList.remove('d-none');
-        input.style.display = '';
-        input.querySelector('input').focus();
+      form.querySelectorAll('.js-auto-score-value-input').forEach((inputDiv) => {
+        inputDiv.classList.remove('d-none');
+        inputDiv.style.display = '';
+        inputDiv.querySelectorAll('input').forEach((input) => {
+          input.name = input.dataset.hiddenInputName;
```

Comment:
This is a slight changed behaviour to account for the Zod validation. In essence, the original code in the server would use `req.body.score_auto_points || null` on the understanding that, if the pencil was never clicked, the `score_auto_points` value would not be sent, which was a mistake, since the input exists but is hidden. This never caused an issue before because the `score_auto_points` is populated with the original auto-points, meaning that manual grading would be set with the auto-points that already existed, so manual grading would typically update it with its current value instead of keeping it as it is (in essence, setting it to the value it already had).

The problem happens when trying to change the auto points from non-zero to zero. Before, this would work because the input would provide `'0'`, which is truthy, but after Zod validation this is now parsed to `0`, which is falsy.

The solution here is that the auto-points input is initially created without a name, but with a `data-hidden-input-name` instead, and when it is enabled the name is set as needed. This will cause the original intended functionality, which is that if pencil is not selected, `score_auto_points` is never sent as a body parameter, and can then be handled as null instead of falsy.